streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImplTest.java

```diff
@@ -117,20 +139,52 @@ public void before() {
         stateDirectory = new StateDirectory(streamsConfig, time, true);
         consumer = new MockConsumer<>(OffsetResetStrategy.NONE);
         stateManager = new GlobalStateManagerImpl(
-            new LogContext("test"),
-            topology,
-            consumer,
-            stateDirectory,
-            stateRestoreListener,
-            streamsConfig);
+                new LogContext("test"),
+                topology,
+                consumer,
+                stateDirectory,
+                stateRestoreListener,
+                streamsConfig);
         processorContext = new InternalMockProcessorContext(stateDirectory.globalStateDir(), streamsConfig);
         stateManager.setGlobalProcessorContext(processorContext);
         checkpointFile = new File(stateManager.baseDir(), ProcessorStateManager.CHECKPOINT_FILE_NAME);
     }
 
     @After
     public void after() throws IOException {
-        stateDirectory.unlockGlobalState();
+        if (stateDirectory != null) {
+            stateDirectory.unlockGlobalState();
+        }
+    }
+
+    public byte[] longToBytes(final long x) {
+        final ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
+        buffer.putLong(x);
+        return buffer.array();
+    }
+
+    @Test(expected = StreamsException.class)
+    public void shouldThrowStreamsExceptionWhenRestoringWithLogAndFailExceptionHandler() {
+        final HashMap<TopicPartition, Long> startOffsets = new HashMap<>();
+        startOffsets.put(t5, 1L);
+        final HashMap<TopicPartition, Long> endOffsets = new HashMap<>();
+        endOffsets.put(t5, 3L);
+        consumer.updatePartitions(t5.topic(), Collections.singletonList(new PartitionInfo(t5.topic(), t5.partition(), null, null, null)));
+        consumer.assign(Collections.singletonList(t5));
+        consumer.updateEndOffsets(endOffsets);
+        consumer.updateBeginningOffsets(startOffsets);
+        final byte[] specialString = "specialKey".getBytes(StandardCharsets.UTF_8);
+        final byte[] longValue = longToBytes(1);
+
+        consumer.addRecord(new ConsumerRecord<>(t5.topic(), t5.partition(), 1, longValue, longValue));
+        consumer.addRecord(new ConsumerRecord<>(t5.topic(), t5.partition(), 2, specialString, specialString));
+        consumer.addRecord(new ConsumerRecord<>(t5.topic(), t5.partition(), 3, longValue, longValue));
+
+        stateManager.initialize();
+        stateManager.register(store5, stateRestoreCallback);
```

Comment:
Somewhat of nitpick but since this test is expected to fail maybe we can get rid of line 186.

Also, we've been moving away from declaring expected exceptions and using a form like 

```java
try {
      stateManager.register(store5, stateRestoreCallback);
      fail("Should not get here as LogAndFailExceptionHandler used");
   } catch (StreamsException e) {
            //expected ok to ignore
  }
```
Might be subjective on my part but the logic is little easier to follow

Reply:
@bbejeck fixed and pushed, thanks for the feedback